managed-ledger/src/test/java/org/apache/zookeeper/MockZooKeeper.java

```diff
@@ -170,27 +172,28 @@ public synchronized void create(final String path, final byte[] data, final List
             return;
         }
 
-        executor.execute(() -> {
+        callbackExecutor.execute(() -> {
             String parent = path.substring(0, path.lastIndexOf("/"));
 
-            synchronized (MockZooKeeper.this) {
-                if (getProgrammedFailStatus()) {
-                    cb.processResult(failReturnCode.intValue(), path, ctx, null);
-                } else if (stopped) {
-                    cb.processResult(KeeperException.Code.CONNECTIONLOSS.intValue(), path, ctx, null);
-                } else if (tree.containsKey(path)) {
-                    cb.processResult(KeeperException.Code.NODEEXISTS.intValue(), path, ctx, null);
-                } else if (!parent.isEmpty() && !tree.containsKey(parent)) {
-                    cb.processResult(KeeperException.Code.NONODE.intValue(), path, ctx, null);
-                } else {
+            if (getProgrammedFailStatus()) {
+                cb.processResult(failReturnCode.intValue(), path, ctx, null);
+            } else if (stopped) {
+                cb.processResult(KeeperException.Code.CONNECTIONLOSS.intValue(), path, ctx, null);
+            } else if (tree.containsKey(path)) {
```

Comment:
Yes, as we can see thread-dump in #434, deadlock is caused by [zkCallback](https://github.com/yahoo/pulsar/blob/master/managed-ledger/src/test/java/org/apache/zookeeper/MockZooKeeper.java#L187) with mutext taken.